tests/test_models_encodings.py

```diff
@@ -125,17 +125,22 @@ def test_unary_encoder(backend, nqubits, architecture, kind):
     # sampling random data in interval [-1, 1]
     sampler = np.random.default_rng(1)
     data = 2 * sampler.random(nqubits) - 1
-    data = backend.cast(data, dtype=data.dtype)
+    data = data.tolist() if kind is not None else backend.cast(data, dtype=data.dtype)
 
-    if kind is not None:
-        data = kind(data)
+    print(type(data), type(data[0]))
+
+    # if kind is not None:
+    #     data = kind(data)
```

Comment:
I'm not sure I understand what is happening here, but I guess it is also work-in-progress.

Reply:
The function needs to be tested on inputs that are either `list` or an `array`. The print is just part of the work in progress.

Reply:
Ok. I see.

Often, in this kind of situations, you can use the property of `np.array()` (and similar, I guess) being idempotent. So, if you want to ensure a list, you first ensure it is an array, i.e., `np.array(data).tolist()`.
I would discourage this in the package code, because it is much better to have clear input types, rather than converting them everywhere (reduce bloating), but for tests is alright.

(I was not commenting the `print()`, it was a marker that was work-in-progress, as the commented code, but that is perfectly fine in a draft)

Reply:
> Ok. I see.
> 
> Often, in this kind of situations, you can use the property of `np.array()` (and similar, I guess) being idempotent. So, if you want to ensure a list, you first ensure it is an array, i.e., `np.array(data).tolist()`. I would discourage this in the package code, because it is much better to have clear input types, rather than converting them everywhere (reduce bloating), but for tests is alright.
> 
> (I was not commenting the `print()`, it was a marker that was work-in-progress, as the commented code, but that is perfectly fine in a draft)

It is not part of the code, just the tests because of how the input `data` is generated.